File: wiles-base/src/main/java/wiles/parser/statements/expressions/InsideMethodCallExpression.java
package wiles.parser.statements.expressions;

import org.jetbrains.annotations.NotNull;
import wiles.parser.builders.ParserContext;
import wiles.parser.services.PrecedenceProcessor;
import wiles.parser.statements.TokenStatement;
import wiles.shared.AbstractCompilationException;
import wiles.shared.Token;

import static wiles.parser.builders.ExpectParamsBuilder.tokenOf;
import static wiles.shared.constants.Tokens.ASSIGN_ID;

public class InsideMethodCallExpression extends AbstractExpression{

    protected boolean isAssignment=false;

    public InsideMethodCallExpression(@NotNull ParserContext oldContext) {
        super(oldContext.setWithinInnerExpression(true));
    }
    {
        isInner = true;
    }

    protected void setComponents(@NotNull PrecedenceProcessor precedenceProcessor) {
        if(isAssignment)
            this.left = precedenceProcessor.getResult();
        else super.setComponents(precedenceProcessor);
    }

    @Override
    protected boolean handleToken(@NotNull Token token) throws AbstractCompilationException {
        if (token.getContent().equals(ASSIGN_ID)) {
            operation = new TokenStatement(transmitter.expect(tokenOf(ASSIGN_ID)), getContext());
            var new_right = new InnerDefaultExpression(getContext());
            exceptions.addAll(new_right.process());
            right = new_right;
            isAssignment = true;
            return true;
        }
        return super.handleToken(token);
    }
}

File: wiles-base/src/main/java/wiles/parser/statements/expressions/InsideMethodCallExpression.kt
package wiles.parser.statements.expressions

import wiles.parser.builders.ExpectParamsBuilder.Companion.tokenOf
import wiles.parser.builders.ParserContext
import wiles.parser.services.PrecedenceProcessor
import wiles.parser.statements.TokenStatement
import wiles.shared.WilesException
import wiles.shared.Token
import wiles.shared.constants.Tokens.ASSIGN_ID

class InsideMethodCallExpression(oldContext: ParserContext) : AbstractExpression(oldContext) {
    private var isAssignment: Boolean = false

    override fun setComponents(precedenceProcessor: PrecedenceProcessor) {
        if (isAssignment) this.left = precedenceProcessor.getResult()
        else super.setComponents(precedenceProcessor)
    }

    @Throws(WilesException::class)
    override fun handleToken(token: Token): Boolean {
        if (token.content == ASSIGN_ID) {
            operation = TokenStatement(transmitter.expect(tokenOf(ASSIGN_ID)), context)
            val newRight = DefaultExpression(context)
            exceptions.addAll(newRight.process())
            right = newRight
            isAssignment = true
            return true
        }
        return super.handleToken(token)
    }
}
